contracts/stake-cw20/src/contract.rs

```diff
@@ -67,42 +75,61 @@ pub fn execute(
         ExecuteMsg::Receive(msg) => execute_receive(deps, env, info, msg),
         ExecuteMsg::Unstake { amount } => execute_unstake(deps, env, info, amount),
         ExecuteMsg::Claim {} => execute_claim(deps, env, info),
-        ExecuteMsg::UpdateConfig { admin, duration } => {
-            execute_update_config(info, deps, admin, duration)
-        }
+        ExecuteMsg::UpdateConfig {
+            owner,
+            manager,
+            duration,
+        } => execute_update_config(info, deps, owner, manager, duration),
+        ExecuteMsg::AddHook { addr } => execute_add_hook(deps, env, info, addr),
+        ExecuteMsg::RemoveHook { addr } => execute_remove_hook(deps, env, info, addr),
     }
 }
 
 pub fn execute_update_config(
     info: MessageInfo,
     deps: DepsMut,
-    new_admin: Option<String>,
+    new_owner: Option<String>,
+    new_manager: Option<String>,
     duration: Option<Duration>,
 ) -> Result<Response, ContractError> {
+    let new_owner = match new_owner {
+        Some(a) => Some(deps.api.addr_validate(&*a)?),
+        None => None,
+    };
+    let new_manager = match new_manager {
+        Some(a) => Some(deps.api.addr_validate(&*a)?),
+        None => None,
+    };
     let mut config: Config = CONFIG.load(deps.storage)?;
-    match config.admin {
-        None => Err(ContractError::NoAdminConfigured {}),
-        Some(current_admin) => {
-            if info.sender != current_admin {
-                return Err(ContractError::Unauthorized {
-                    expected: current_admin,
-                    received: info.sender,
-                });
-            }
+    if Some(info.sender.clone()) != config.owner && Some(info.sender.clone()) != config.manager {
```

Comment:
Using pattern matching would make this code a lot simpler

Reply:
I am not sure pattern matching makes it that much simpler, happy to take a look at a suggested change.